keyserver/listener.go

```diff
@@ -55,21 +54,24 @@ func (server *ConiksServer) acceptClient(conn net.Conn, handler func(msg []byte)
 
 	var buf bytes.Buffer
 	if _, err := io.CopyN(&buf, conn, 8192); err != nil && err != io.EOF {
-		log.Printf("client read %v: %v", conn.RemoteAddr(), err)
+		server.logger.Errorw(err.Error(),
+			"address", conn.RemoteAddr().String())
 		return
 	}
 
 	res, err := handler(buf.Bytes())
 	// TODO: The `err` returned here is purely for logging purposes.  It
 	// would be better for `handler` not to return any error, and instead
 	// log if the error code in the `res` is not `ReqSuccess`.
-	if err != nil {
-		log.Printf("client handle %v: %v", conn.RemoteAddr(), err)
+	if err != ReqSuccess {
+		server.logger.Warnw(err.Error(),
+			"address", conn.RemoteAddr().String())
 	}
 
 	_, err = conn.Write([]byte(res))
 	if err != nil {
-		log.Printf("client write %v: %v", conn.RemoteAddr(), err)
+		server.logger.Errorw(err.Error(),
```

Comment:
I think we do not expect an error here (hence `Error`), whereas the above `err != ReqSuccess` is the return status and if there is something like `ErrDirectory` etc I think we would like to treat it as a warning.